src/importexport/musicxml/internal/musicxml/import/importmusicxmlpass2.cpp

```diff
@@ -6006,2924 +5825,3007 @@ void MusicXmlParserPass2::divisions()
-        int fret = notation.text().toInt();
-        if (note) {
-            if (note->staff()->isTabStaff(Fraction(0, 1))) {
-                note->setFret(fret);
-            }
-        } else {
-            m_logger->logError(u"no note for fret", &m_e);
-        }
-    } else if (notation.name() == "pluck") {
-        addTextToNote(m_e.lineNumber(), m_e.columnNumber(), notation.text(), placement, fontWeight, fontSize, fontStyle, fontFamily,
-                      color, TextStyleType::RH_GUITAR_FINGERING, m_score, note);
-    } else if (notation.name() == "string") {
-        if (note) {
-            if (note->staff()->isTabStaff(Fraction(0, 1))) {
-                note->setString(notation.text().toInt() - 1);
-            } else {
-                addTextToNote(m_e.lineNumber(), m_e.columnNumber(), notation.text(), placement, fontWeight, fontSize, fontStyle, fontFamily,
-                              color, TextStyleType::STRING_NUMBER, m_score, note);
-            }
-        } else {
-            m_logger->logError(u"no note for string", &m_e);
-        }
-    }
+    return m_fretDiagram ? m_fretDiagram->visible() : false;
 }
 
-void MusicXmlParserNotations::arpeggio()
+MusicXmlParserNote::MusicXmlParserNote(muse::XmlStreamReader& e, engraving::Score* score, MusicXmlLogger* logger,
+                                       MusicXmlParserPass1& pass1,
+                                       MusicXmlParserPass2& pass2, const muse::String& partId, engraving::Measure* measure,
+                                       const Fraction sTime, const Fraction prevSTime, Fraction& missingPrev, Fraction& dura,
+                                       Fraction& missingCurr, String& currentVoice, GraceChordList& gcl, size_t& gac, Beams& currBeams,
+                                       FiguredBassList& fbl, int& alt, MusicXmlTupletStates& tupletStates, Tuplets& tuplets,
+                                       ArpeggioMap& arpMap, DelayedArpMap& delayedArps)
+    : m_e(e), m_score(score), m_logger(logger), m_pass1(pass1), m_pass2(pass2), m_partId(partId), m_measure(measure), m_sTime(sTime),
+    m_prevSTime(prevSTime), m_missingPrev(missingPrev), m_durationFraction(dura), m_missingCurr(missingCurr), m_currentVoice(currentVoice),
+    m_graceChordsList(gcl), m_graceAfterCount(gac), m_currBeams(currBeams), m_figuredBassList(fbl), m_alter(alt), m_tupletStates(
+        tupletStates), m_tuplets(tuplets), m_arpMap(arpMap), m_delayedArps(delayedArps), m_lyricParser(m_pass1.getMusicXmlPart(
+                                                                                                           m_partId).lyricNumberHandler(), m_e, m_score, m_logger, m_pass1.isVocalStaff(
+                                                                                                           m_partId)), m_notationsParser(
+        m_e, m_score, m_logger,
+        m_pass2),
+    m_noteDurationParser(m_pass2.divs(), m_logger, &m_pass1), m_notePitchParser(m_logger)
```

Comment:
This bit is very difficult to read. Would be slightly better to put every member on a separate line.